frontend/src/features/tasks/hooks/useUnifiedMessages.ts

```
@@ -218,16 +218,37 @@ export function useUnifiedMessages({
+  // Track which tasks have had their initial recovery to avoid skipping
+  // recovery when chat:start races ahead and sets isStreaming=true before recover() runs
+  const recoveredTasksRef = useRef<Set<number>>(new Set())
+
+  // Cleanup old recovered task IDs to prevent unbounded growth
+  // Keep only the most recent MAX_RECOVERED_TASKS entries
+  const cleanupRecoveredTasks = useCallback(() => {
+    const MAX_RECOVERED_TASKS = 50
+    const tasks = Array.from(recoveredTasksRef.current)
+    if (tasks.length > MAX_RECOVERED_TASKS) {
+      const toRemove = tasks.slice(0, tasks.length - MAX_RECOVERED_TASKS)
+      toRemove.forEach(id => recoveredTasksRef.current.delete(id))
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)